concordium-consensus/src/Concordium/KonsensusV1/TreeState/Implementation.hs

```diff
@@ -800,11 +815,12 @@ finalizeTransactions = mapM_ removeTrans
         transactionTable
             . TT.ttNonFinalizedTransactions
             . at' sender
-            ?=! ( anft
+            . non TT.emptyANFT
+            .=! ( anft
```

Comment:
The `non` here will not remove the empty value, since the `anftNextNonce` is not `minNonce`, which is what `emptyANFT` has.

I suggest being a bit more verbose and explicit in writing out the logic.